<?xml version="1.0" encoding="UTF-8"?>

<launch>
    <arg name = "vo_topic"  default = "/frontend/vo"/>
    <!-- <arg name = "stereo_cloud_topic" default = "/frontend/point_cloud"/> -->
    <!-- <arg name = "gnss_topic" default = "/kitti/oxts/gps/fix"/> -->
    <arg name = "gnss_topic" default = "/tf"/>
    <arg name = "imu_topic" default = "/kitti/oxts/imu"/>

    <node 
        name   = "backend"
        pkg    = "backend"
        type   = "backend_node"
        args   = "$(arg vo_topic) 
                  $(arg gnss_topic) 
                  $(arg imu_topic)"
        output = "screen"
    >
    
    <remap from = "vo_topic"  to = "$(arg vo_topic)"/>
    <remap from = "gnss_topic"  to = "$(arg gnss_topic)"/>
    <remap from = "imu_topic" to = "$(arg imu_topic)"/>
    </node>

</launch>